src/fr-device/index.ts

```diff
@@ -0,0 +1,209 @@
+import {
+  categories,
+  delay,
+  fontNames,
+  displayProps,
+  browserProps,
+  hardwareProps,
+  platformProps,
+} from './defaults';
+import {
+  Category,
+  CollectParameters,
+  DeviceProfileData,
+  Geolocation,
+  ProfileConfigOptions,
+} from './interfaces';
+
+/**
+ * @class FRDevice - Collects user device metadata
+ */
+class FRDevice {
```

Comment:
Profile collection class for generating the Device Profile. This is the primary code for this feature.